ui/lib/core/addon/components/search-select.js

```diff
@@ -1,99 +1,129 @@
- * @param {boolean} [displayInherit] - if you need the search select component to display inherit instead of box.
+ * @param {boolean} [displayInherit=false] - if you need the search select component to display inherit instead of box.
  * @param {boolean} [renderInfoTooltip=false] - if you want search select to render a tooltip beside a selected item if no corresponding model was returned from .query
  *
- * @param {Array} options - *Advanced usage* - `options` can be passed directly from the outside to the
- * power-select component. If doing this, `models` should not also be passed as that will overwrite the
- * passed value. ex: [{ name: 'namespace45', id: 'displayedName' }];
- * @param {function} search - *Advanced usage* - Customizes how the power-select component searches for matches -
- * see the power-select docs for more information.
+ // * advanced customization
+ * @param {Array} options - array of objects passed directly to the power-select component. If doing this, `models` should not also be passed as that will overwrite the
+ * passed options. ex: [{ name: 'namespace45', id: 'displayedName' }];
+ * @param {function} search - Customizes how the power-select component searches for matches - see the power-select docs for more information.
  *
  */
-export default Component.extend({
-  layout,
-  'data-test-component': 'search-select',
-  attributeBindings: ['data-test-component'],
-  classNameBindings: ['displayInherit:display-inherit'],
-  classNames: ['field', 'search-select'],
-  store: service(),
-  onChange: () => {},
-  inputValue: computed(function () {
-    return [];
-  }),
-  allOptions: null, // list of options including matched
-  selectedOptions: null, // list of selected options
-  options: null, // all possible options
-  shouldUseFallback: false,
-  shouldRenderName: false,
-  disallowNewItems: false,
-  passObject: false,
-  objectKeys: null,
-  idKey: computed('objectKeys', function () {
-    // if objectKeys exists, then use the first element of the array as the identifier
-    return this.objectKeys ? this.objectKeys[0] : 'id';
-  }),
-  init() {
-    this._super(...arguments);
-    this.set('selectedOptions', this.inputValue || []);
-  },
-  didRender() {
-    this._super(...arguments);
-    let { oldOptions, options, selectedOptions } = this;
-    let hasFormattedInput = typeof selectedOptions.firstObject !== 'string';
-    if (options && !oldOptions && !hasFormattedInput) {
-      // this is the first time they've been set, so we need to format them
-      this.formatOptions(options);
+
+export default class SearchSelect extends Component {
+  @service store;
+  @tracked selectedOptions; // array of selected options, initially array of strings, then array of objects **WHYY/WHEN?!**
+  @tracked allOptions; // array id strings for ALL existing options, including matched
+  @tracked searchSelectOptions; // list of options rendered in dropdown
+  @tracked shouldUseFallback = false;
+
+  get disallowNewItems() {
```

Comment:
Kept the default `false` per the original component. 

@ivana-mcconnell - we discussed last UI sync to change this so that SearchSelect _does not_ allowing "creating" new items by default, right? As it's confusing since you're not actually creating the thing, just associating the name of the thing. Should I go ahead and make that change?

Reply:
That's right— remove the ability to create new things here, since we can't (and won't) show the full database connection form here.

Reply:
Discussed with Ivana on slack and we decided to hold off making this change for now